packages/parser/tests/sankey.test.ts

```diff
@@ -0,0 +1,194 @@
+import { describe, expect, it } from 'vitest';
+import type { LangiumParser, ParseResult } from 'langium';
+
+import type { SankeyServices } from '../src/language/index.js';
+import { Sankey, createSankeyServices } from '../src/language/index.js';
+import { noErrorsOrAlternatives } from './test-util.js';
+
+const services: SankeyServices = createSankeyServices().Sankey;
+const parser: LangiumParser = services.parser.LangiumParser;
+export function createSankeyTestServices(): {
+  services: SankeyServices;
+  parse: (input: string) => ParseResult<Sankey>;
+} {
+  const parse = (input: string) => {
+    return parser.parse<Sankey>(input);
+  };
+  return { services, parse };
+}
+
+describe('sankey', () => {
+  const { parse } = createSankeyTestServices();
+
+  it('should handle simple sankey', () => {
+    const context = `sankey-beta
+    sourceNode, targetNode, 10`;
+    const result = parse(context);
+    noErrorsOrAlternatives(result);
+
+    const value = result.value;
+    expect(value.$type).toBe(Sankey);
+    expect(value.links[0].source).toBe('sourceNode');
+    expect(value.links[0].target).toBe('targetNode');
+    expect(value.links[0].value).toBe(10);
+
+    expect(value.nodes).toStrictEqual(['sourceNode', 'targetNode']);
+  });
+
+  it('should handle sankey with double quotes', () => {
+    const context = `sankey-beta
+"source node, with comma","target node, with comma","10.00"
+    `;
+    const result = parse(context);
+    noErrorsOrAlternatives(result);
+
+    const value = result.value;
+    expect(value.$type).toBe(Sankey);
+    expect(value.links[0].source).toBe('source node, with comma');
+    expect(value.links[0].target).toBe('target node, with comma');
+    expect(value.links[0].value).toBe(10);
+
+    expect(value.nodes).toStrictEqual(['source node, with comma', 'target node, with comma']);
+  });
+
+  it('should handle sankey with double quotes with newline and doable quotes', () => {
+    const context = `sankey-beta
+"source node
+"" with comma","target node
+"" with comma","10.00"
+    `;
+    const result = parse(context);
+    noErrorsOrAlternatives(result);
+
+    const value = result.value;
+    expect(value.$type).toBe(Sankey);
+    expect(value.links[0].source).toBe(`source node
+" with comma`);
+    expect(value.links[0].target).toBe(`target node
+" with comma`);
```

Comment:
I know this is simply a copy of previous example and not a big deal, nevertheless there is no comma, there is a quote.

Reply:
done.